### ISSUE TITLE
Error for cuda and cpu

### ISSUE THREAD

**Issue by lucasjinreal** [IBM/pytorch-seq2seq#185]
```
/pytorch-seq2seq/seq2seq/models/EncoderRNN.py", line 68, in forward
    embedded = self.embedding(input_var)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 479, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/sparse.py", line 113, in forward
    self.norm_type, self.scale_grad_by_freq, self.sparse)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py", line 1283, in embedding
    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected object of backend CUDA but got backend CPU for argument #3 'index'

```

**Comment by pskrunner14** [comment]
@jinfagang could you try running your code with `develop` branch?

**Comment by xiaodaoyoumin** [comment]
bug bug bug  !!!!!!!!!!!!!!!!!!!!!!!!!!!!   waste my time!!!!!!!!!!!

**Comment by lucasjinreal** [comment]
@Hujun-Cui I think this repo is not active maintained now.

**Comment by pskrunner14** [comment]
@jinfagang this has been a recurring issue, however the fix has already been made on the `develop` branch. Will update on `master` soon. Please see #180 for a simple workaround.

**Comment by v587su** [comment]
The wrong code is at line 75 of supervised_trainer.py and line 38 of evaluator.py. 
I change them to  `device = torch.device('cuda:0') if torch.cuda.is_available() else -1` and CUDA is available now.

**Comment by rain1024** [comment]
I think this should be merged in master now.

**Comment by Hexagram-King** [comment]
> The wrong code is at line 75 of supervised_trainer.py and line 38 of evaluator.py.
> I change them to `device = torch.device('cuda:0') if torch.cuda.is_available() else -1` and CUDA is available now.

Except the two files, is there any other file need to fix? After fix the two files, the code still be incorrect...

**Comment by isekulic** [comment]
> > The wrong code is at line 75 of supervised_trainer.py and line 38 of evaluator.py.
> > I change them to `device = torch.device('cuda:0') if torch.cuda.is_available() else -1` and CUDA is available now.
> 
> Except the two files, is there any other file need to fix? After fix the two files, the code still be incorrect...

No, but you need to re-install the package after you do these corrections. Running `python setup.py install` again helped in my case.
